### PR TITLE
feat: add onDateSelected to know if component changes period

### PR DESCRIPTION
**Pull request by marcossaore**
✨ ** New feature**

**Why these changes?**
I added a new function `onDateSelected` to handle previous and next date selection events in the calendar. It is important to know if component has a new period to select.
**My context:**
I will make request to the server and when it returns, I want to disable some dates but the component does not provided a function to inform what the period is showing.

**How do we test?**
1. Navigate to the calendar component, put `onDateSelected={onDateSelected}` in component.
2. Interact with the calendar and click on previous or next buttons.
3. Observe if the `onDateSelected` function is triggered appropriately.
4. I do not create test because the function onDayClick not have one. And I had need to some example to do it.

### CODE REVIEW

**Review by dysfunc — CHANGES_REQUESTED** [review]
I made suggestions to support multipane pickers, since your end period should be the last day of the last visible month.  I also adjusted the nomenclature for this to be more specific/explicit to what is happening. I felt like `onDateChange` and `onDateSelected` are misleading since we aren't selecting or changing the actual selected date. Thoughts?

File: src/datepicker.svelte [hunk 1]
```diff
@@ -334,6 +340,26 @@
...
+    onDateSelected({
+      action,
+      actionPressed,
+      selectedPeriod: {
+        start: `${startDateYear}-${month >= 10 ? month : `0${month}`}-${initialDayOffMonth}`,
+        end: `${startDateYear}-${month >= 10 ? month : `0${month}`}-${lastDayOfMonth}`
+      },
+      isPastePeriod: selectedPeriod < initialPeriod
+    })
+  }
```
[the remark this answers is not in the document]
Comment (marcossaore) [reply src/datepicker.svelte:361]: I agree, I couldn't think of a better name, this is great.
Have you ever thought about having a way to disable the hours or some interval between them?

File: package-lock.json
[the remark this answers is not in the document]
Comment (marcossaore) [reply package-lock.json:1]: for app and docs?

[the remark this answers is not in the document]
Comment (dysfunc) [reply package-lock.json:1]: Please delete it from the root directory. `docs` should container a lock file

File: src/datepicker.svelte [hunk 2]
```diff
@@ -334,6 +340,44 @@
...
+   * @param {string} type - The type of navigation (month or year).
+   */
+  const onNavigation = async (direction, type) => {
+    await tick();
+
+    const initial = new Date(defaultYear, defaultMonth);
+    const initialDayOffMonth = '01';
+
+    let current = new Date(startDateYear, startDateMonth);
+    let year = isMultipane ? endDateYear : startDateYear;
```
Comment (dysfunc) [inline src/datepicker.svelte:355]:
```suggestion
```

Comment (dysfunc) [reply src/datepicker.svelte:355]: I forgot to remove this

File: package-lock.json
[the remark this answers is not in the document]
Comment (marcossaore) [reply package-lock.json:1]: do it.

**Pull request merged**